cmd/iam.go

```diff
@@ -1664,15 +1679,30 @@ func (sys *IAMSys) IsAllowedSTS(args iampolicy.Args) bool {
 		// No policy available reject.
 		return false
 	}
-	name := mp.Policy
 
-	if pnameSlice[0] != name {
+	if strings.Join(policies, ",") != mp.Policy {
```

Comment:
We should treat it as a set of policies rather than a slice because the ordering is not relevant and this comparison can fail for that reason.

Reply:
Yes makes sense.